### PR TITLE
xUnit test migration (Tester project)

### PR DESCRIPTION
**Pull request by galvesribeiro**
This PR has the following goals:

1. Remove the MSTest runner from **Tester** project and replace it with xunit.runner
2. Keep MSTest assertion APIs for now so there will be no changes on the assertion and test logic so far
3. Keep the test code with the minimal changes as possible.
4. Make the *Tester*  project to run on Jenking/CI while keeping **TesterInternal** on MSTests.
5. Move one step forward to #368 by removing the dependency at some point of MSTest and adopting xunit which is the most supported test runner for DNX/CoreCLR and is what ASP.Net and CoreCLR teams are already using on it.

After this PR get Merged we will get 2 more steps:

1. Migrate **TesterInternal** to xunit.runner
2. Migrate the assertion APIs to something else but MSTest ones

### CODE REVIEW

**Comment by jdom** [comment]
Helped @galvesribeiro rebase on top of the latest and polished a little bit. It should be ready to merge, assuming Jenkins runs it fine. Feedback is welcome from others at this point, as we don't want this massive PR to linger, as it always has non-trivial merge conflicts (due to the move of the Tester project to a new location)

File: test/Tester/GrainActivateDeactivateTests.cs [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
 namespace UnitTests.ActivationsLifeCycleTests
 {
-    [TestClass]
     public class GrainActivateDeactivateTests : HostedTestClusterEnsureDefaultStarted
     {
         private IActivateDeactivateWatcherGrain watcher;
-
-        [TestInitialize]
+        
         public void TestInitialize()
```
Comment (jdom) [inline test/Tester/GrainActivateDeactivateTests.cs]: this should be in the constructor

File: test/Tester/GrainActivateDeactivateTests.cs [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
         {
             watcher = GrainClient.GrainFactory.GetGrain<IActivateDeactivateWatcherGrain>(0);
             watcher.Clear().Wait();
         }
-
-        [TestCleanup]
+        
         public void TestCleanup()
```
Comment (jdom) [inline test/Tester/GrainActivateDeactivateTests.cs]: and this on Dispose

File: test/Tester/StreamingTests/PullingAgentManagementTests.cs [hunk 3]
```diff
@@ -12,18 +13,15 @@
...
-    [TestClass]
-    public class PullingAgentManagementTests : HostedTestClusterPerFixture
+    public class PullingAgentManagementTests : OrleansTestingBase, IDisposable
     {
         private const string adapterName = StreamTestsConstants.AZURE_QUEUE_STREAM_PROVIDER_NAME;
         private readonly string adapterType = typeof(AzureQueueStreamProvider).FullName;
+        private readonly TestingSiloHost HostedCluster;
 
-        public static TestingSiloHost CreateSiloHost()
+        public PullingAgentManagementTests()
```
Comment (jdom) [inline test/Tester/StreamingTests/PullingAgentManagementTests.cs]: this used to be HostedTestClusterPerFixture, but now it's restarting per test. I'm fixing it now

File: test/Tester/StreamingTests/SampleStreamingTests.cs [hunk 4]
```diff
@@ -33,63 +26,83 @@ public static TestingSiloHost CreateSiloHost()
...
+        private string streamProvider;
+        
+        private string deploymentId;
+
+        public SampleStreamingTests(SampleStreamingTestsFixture fixture)
+        {
+            deploymentId = fixture.HostedCluster.DeploymentId;
+        }
+        
         public void TestCleanup()
```
Comment (jdom) [inline test/Tester/StreamingTests/SampleStreamingTests.cs]: Should do this on dispose

**Comment by mmitche** [comment]
@dotnet-bot test ci please

**Comment by mmitche** [comment]
@dotnet-bot test ci please

**Comment by mmitche** [comment]
@dotnet-bot test ci please

**Comment by mmitche** [comment]
LGTM on the CI changes.  Job looks correct.

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by jdom** [comment]
@dotnet-bot test this please

**Comment by jdom** [comment]
@dotnet-bot test this please

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by jdom** [comment]
@dotnet-bot test ci please

**Comment by jdom** [comment]
@dotnet-bot test this please

**Comment by jdom** [comment]
This is finally in! Thanks @galvesribeiro! It took a long time to sort out the build issues in the many systems we use, but it's finally here. Let's sync up tomorrow on learnings so that we can do TesterInternal at some point too

**Comment by sergeybykov** [comment]
Another mega effort completed!

Notice the same pattern - driven by the community with help from the core team (mostly due to the VSO tax).

Well done, @galvesribeiro, @jdom and others who helped!

**Comment by gabikliot** [comment]
Great effort indeed!

I thought we moved all unit test from VSO to GH? Only stress tests are still there, but they were not using same framework, as far as I remember. Is it more complicated?

**Comment by jdom** [comment]
Unfortunately there's still some in VSO only, and also we use VSO to run functionals against the real service dependencies, such as azure storage or eventhub

**Comment by gabikliot** [comment]
:-1:

**Comment by galvesribeiro** [comment]
Thanks for the help @jdom. Lets move forward and do TestInternals.

**Comment by ZhaoXiangXML** [comment]
I'm trying to write my own functional test for Orleans and .NET Core, but it does not work, would you please check it out on #2846?

**Pull request merged**